Microsoft.Health.Dicom.CosmosDb/CosmosDataStore.Retrieve.cs

```diff
@@ -11,34 +11,66 @@
 
 //using FellowOakDicom;
 
+using Microsoft.Azure.Cosmos;
 using Microsoft.Health.Dicom.Core.Features.Model;
+using Microsoft.Health.Dicom.Core.Features.Query;
 
 namespace Microsoft.Health.Dicom.CosmosDb
 {
     public partial class CosmosDataStore
     {
-        private async Task<IEnumerable<DataField>> GetItems()
+        private async Task<IEnumerable<DataField>> GetItems(IReadOnlyCollection<QueryFilterCondition> filterConditions)
         {
-            //var data = new DataField() { Value = new { val = 45 } };
-            var query = "SELECT * FROM c WHERE c['value']['00080020']['Value'][0] = '20200922'";
-            //var query = "SELECT * FROM c";
-            var res1 = Container.GetItemQueryIterator<DataField>(query);
+            CosmosDbQueryGenerator cosmosDbQueryGenerator = new CosmosDbQueryGenerator();
+            var maxItemCount = 100;
+            var query = "";
+            foreach (var condition in filterConditions)
+            {
+                condition.Accept(cosmosDbQueryGenerator);
+            }
+
+            // FUTURE TODO: continuation tokens ? will need a POST for a long token
+            query = $"SELECT * FROM c WHERE {cosmosDbQueryGenerator.OutputQuery()}";
```

Comment:
See if you can surrond the `WHERE` conidtion with a parentheses like this
`SELECT * FROM c WHERE ({cosmosDbQueryGenerator.OutputQuery()})`
That way if the expression spans multiple logical expressions the precedence is clear